<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>MSEA enrichment - single cell metabolomics • S2IsoMEr</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="MSEA enrichment - single cell metabolomics">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">S2IsoMEr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/MSEA_enrichment-single_cell_metabolomics.html">MSEA enrichment - single cell metabolomics</a></li>
    <li><a class="dropdown-item" href="../articles/ORA_enrichment-single_cell_metabolomics.html">ORA enrichment - single cell metabolomics</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>MSEA enrichment - single cell metabolomics</h1>
                        <h4 data-toc-skip class="author">Bishoy
Wadie</h4>
                        
      

      <div class="d-none name"><code>MSEA_enrichment-single_cell_metabolomics.Rmd</code></div>
    </div>

    
    
<p>Single-cell metabolomics generates output similar to scRNA-Seq,
resulting in a matrix of metabolites (m) by cells (c), where values
represent the abundance of each metabolite per cell. As with any
single-cell analysis pipeline, enrichment analysis is typically one of
the final steps to interpret the differential markers identified
earlier.</p>
<p>While Overrepresentation Analysis (ORA) focuses on the identity of
selected markers for enrichment, it does not account for their
abundance. To incorporate abundance, ranking-based methods are
preferred, with Metabolite Set Enrichment Analysis (MSEA) being the most
widely used enrichment method for both bulk and single-cell
metabolomics. In metabolomics, MSEA tools typically require a list of
compounds and their associated concentrations as input, primarily for
targeted metabolomics. The most common ones are <a href="https://academic.oup.com/nar/article/38/suppl_2/W71/1101310" class="external-link">MSEA</a>
and <a href="https://www.metaboanalyst.ca/" class="external-link">MetaboAnalyst</a>. We
recommend users explore these tools for more information on MSEA.</p>
<p>For MS1-based metabolomics datasets, particularly in imaging MS, the
inherent molecular ambiguity in metabolite identification complicates
downstream analyses, including enrichment. To our knowledge, no other
metabolite enrichment method addresses isomeric/isobaric ambiguity in
MSEA.</p>
<p>In this notebook, we demonstrate how to use <code>S2IsoMEr</code> for
MSEA enrichment in single-cell metabolomics datasets while accounting
for isomeric/isobaric ambiguity. The single-cell dataset used is from
the SpaceM paper, which models NASH by stimulating Hepa-RG cells with
fatty acids and other inhibitors compared to a control, followed by
MALDI imaging MS.</p>
<p>The data is freely available in <a href="https://www.ebi.ac.uk/metabolights/editor/MTBLS78/descriptors" class="external-link">MetaboLights</a>.</p>
<div class="section level2">
<h2 id="loading-packages">Loading packages<a class="anchor" aria-label="anchor" href="#loading-packages"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://alexandrovteam.github.io/S2IsoMEr/">S2IsoMEr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="dataset">Dataset<a class="anchor" aria-label="anchor" href="#dataset"></a>
</h2>
<p>The single-cell dataset used is from the SpaceM paper, which models
NASH by stimulating Hepa-RG cells with fatty acids and other inhibitors
compared to a control, followed by MALDI imaging MS.</p>
<p>The data is freely available in <a href="https://www.ebi.ac.uk/metabolights/editor/MTBLS78/descriptors" class="external-link">MetaboLights</a>.</p>
<div class="section level3">
<h3 id="download-single-cell-matrices-and-associated-metadata">Download single-cell matrices and associated metadata<a class="anchor" aria-label="anchor" href="#download-single-cell-matrices-and-associated-metadata"></a>
</h3>
<p><code>NASH_scm</code> contains the single-cell metabolite matrix
which will be main input as well as condition per cell in
<code>NASH_scm$metadata</code>. These are the main required files to run
single-cell metabolomics enrichment. <code>condition_metadata</code>
contains the <a href="https://metaspace2020.eu/" class="external-link">METASPACE</a> dataset
names for each replicate, while <code>metaspace_annotations</code>
contains the annotation results for each dataset in the <a href="https://metaspace2020.eu/project/Rappez_2021_SpaceM?tab=datasets" class="external-link">SpaceM</a>
project on METASPACE. We will use the annotation results and
corresponding FDR thresholds to select metabolites as input query and
corresponding universe for enrichment.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NASH_scm_tmp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://zenodo.org/records/13318721/files/NASH_scm_dataset.rds"</span>, destfile <span class="op">=</span> <span class="va">NASH_scm_tmp</span><span class="op">)</span></span>
<span><span class="va">NASH_scm</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">NASH_scm_tmp</span><span class="op">)</span></span>
<span></span>
<span><span class="va">condition_metadata_tmp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://zenodo.org/records/13318721/files/spacem_scm_matrices.rds"</span>, destfile <span class="op">=</span> <span class="va">condition_metadata_tmp</span><span class="op">)</span></span>
<span><span class="va">condition_metadata</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">condition_metadata_tmp</span><span class="op">)</span><span class="op">[[</span><span class="st">"metaspace_dataset_names"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="va">metaspace_annotations_tmp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://zenodo.org/records/13318721/files/SpaceM_metaspace_ds_annotations.rds"</span>, destfile <span class="op">=</span> <span class="va">metaspace_annotations_tmp</span><span class="op">)</span></span>
<span><span class="va">metaspace_annotations</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">metaspace_annotations</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="prepare-the-input-data">Prepare the input data<a class="anchor" aria-label="anchor" href="#prepare-the-input-data"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scm</span> <span class="op">=</span> <span class="va">NASH_scm</span><span class="op">$</span><span class="va">scm</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">conds</span> <span class="op">=</span> <span class="va">NASH_scm</span><span class="op">$</span><span class="va">metadata</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">column_to_rownames</span><span class="op">(</span><span class="st">"Cell"</span><span class="op">)</span></span>
<span><span class="va">conds</span> <span class="op">=</span> <span class="va">conds</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">scm</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">conds_unique</span> <span class="op">=</span> <span class="va">conds</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metaspace_annotations</span> <span class="op">=</span> <span class="va">metaspace_annotations</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">condition_metadata</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ds_name"</span> <span class="op">=</span> <span class="st">"dataset_name"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="st">"Replicate"</span> <span class="op">=</span> <span class="st">"Condition"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">conds_unique</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="msea">MSEA<a class="anchor" aria-label="anchor" href="#msea"></a>
</h2>
<p>In this notebook, we will run MSEA for all conditions against the
control. We will follow the same first steps for object initialization
as introduced in the <a href="ORA_enrichment-single_cell_metabolomics.html">ORA vignette</a>,
check it out for more information on the parameters passed to the
<code>initEnrichment</code> function.</p>
<p>Additional arguments to <code>initEnrichment</code> that wasn’t used
for <a href="ORA_enrichment-single_cell_metabolomics.html">ORA</a> but
needed for MSEA include :</p>
<ol style="list-style-type: decimal">
<li>
<p><code>ranking.by</code> : Ranking metric for MSEA, supported
options are :</p>
<ol style="list-style-type: decimal">
<li><p><strong><code>logFC</code> :</strong> Log2 Fold change. Most
common and easier to interpret</p></li>
<li><p><strong><code>wilcox.test</code> :</strong> Wilcoxon rank sum
statistic (signed based on LFC sign to improve interpretation)</p></li>
<li><p><strong><code>t.test</code> :</strong> Student’s t-Test
One-tailed (signed based on LFC sign to improve interpretation)</p></li>
<li><p><strong><code>BWS</code> :</strong> Baumgartner-Weiss-Schindler
test statistic (signed based on LFC sign to improve
interpretation)</p></li>
</ol>
</li>
<li><p><code>gsea.method</code> : MSEA method. Based on either <a href="https://github.com/franapoli/signed-ks-test/blob/master/signed-ks-test.R" class="external-link">KS-signed
method</a> or <a href="https://bioconductor.org/packages/release/bioc/html/fgsea.html" class="external-link">fgsea</a>.
We recommend using <strong>fgsea</strong></p></li>
</ol>
<p>Note : <code>BWS</code> was recommended as an alternative to Wilcoxon
rank sum test in this <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1674-0" class="external-link">paper</a>.
While we support <code>BWS</code> as a ranking metric, we recommend
double checking the interpretation of the results compared to Log2 Fold
changes which are more commonly used.</p>
<p>Following object initialization, we call <code>Run_enrichment</code>
as before which internally calls <code>Run_bootstrap_MSEA</code>. From
the additional list of arguments to <code>Run_bootstrap_MSEA</code> we
recommend defining the following arguments to
<code>Run_enrichment</code> :</p>
<ul>
<li><p><code>n_bootstraps</code>: Number of bootstrap iterations. The
default is 50, but increasing this number generally improves accuracy,
though it will slow down the process. Adjust according to your needs. We
recommend a minimum of 50 and a maximum of 1000. 100 is
acceptable.</p></li>
<li><p><code>min_pathway_size</code> : The minimum number of metabolites
that must be present in a given term for it to be considered</p></li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">desired_fdr</span> <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="va">desired_annot_db</span> <span class="op">=</span> <span class="st">"HMDB"</span></span>
<span></span>
<span><span class="va">Multi_cond_res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">MSEA_results</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">c</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"F"</span>, <span class="st">"FI"</span>, <span class="st">"FIT"</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">cond_x</span> <span class="op">=</span> <span class="st">"U"</span></span>
<span>  <span class="va">cond_y</span> <span class="op">=</span> <span class="va">c</span></span>
<span></span>
<span>  <span class="va">annots_des_fdr</span> <span class="op">=</span> <span class="va">metaspace_annotations</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Condition</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">cond_x</span>, <span class="va">cond_y</span><span class="op">)</span>,</span>
<span>                  <span class="va">fdr</span> <span class="op">&lt;=</span> <span class="va">desired_fdr</span>,</span>
<span>                  <span class="fu">str_detect</span><span class="op">(</span><span class="va">db</span>, <span class="va">desired_annot_db</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">formula_adduct</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scm</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">input_scm</span> <span class="op">=</span> <span class="va">scm</span><span class="op">[</span><span class="va">annots_des_fdr</span>,<span class="op">]</span></span>
<span></span>
<span></span>
<span>  <span class="va">MSEA_obj</span> <span class="op">=</span> <span class="fu"><a href="../reference/initEnrichment.html">initEnrichment</a></span><span class="op">(</span>scmatrix <span class="op">=</span> <span class="va">input_scm</span>, conditions <span class="op">=</span> <span class="va">conds</span><span class="op">$</span><span class="va">Condition</span>,</span>
<span>                            enrichment_type <span class="op">=</span> <span class="st">"MSEA"</span>,annot_db <span class="op">=</span> <span class="va">desired_annot_db</span>,</span>
<span>                            consider_isomers <span class="op">=</span> <span class="cn">T</span>, consider_isobars <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                            polarization_mode <span class="op">=</span> <span class="st">"positive"</span>,</span>
<span>                            background_type <span class="op">=</span> <span class="st">"sub_class"</span>,</span>
<span>                            molecule_type <span class="op">=</span> <span class="st">"Metabo"</span>,</span>
<span>                            condition.x <span class="op">=</span> <span class="va">cond_x</span>,</span>
<span>                            condition.y <span class="op">=</span> <span class="va">cond_y</span>,</span>
<span>                            ranking.by <span class="op">=</span> <span class="st">"BWS"</span>,</span>
<span>                            gsea.method <span class="op">=</span> <span class="st">"fgsea"</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">MSEA_res</span> <span class="op">=</span> <span class="fu"><a href="../reference/Run_enrichment.html">Run_enrichment</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">MSEA_obj</span>,report_ambiguity_scores <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">enrich_res</span> <span class="op">=</span> <span class="va">MSEA_res</span><span class="op">$</span><span class="va">enrichment_analysis</span><span class="op">$</span><span class="va">enrichment_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>condition.x <span class="op">=</span> <span class="st">"U"</span>,</span>
<span>                  condition.y <span class="op">=</span> <span class="va">c</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">Multi_cond_res</span><span class="op">[[</span><span class="va">c</span><span class="op">]</span><span class="op">]</span> <span class="op">=</span> <span class="va">enrich_res</span></span>
<span>  <span class="va">MSEA_results</span><span class="op">[[</span><span class="va">c</span><span class="op">]</span><span class="op">]</span> <span class="op">=</span> <span class="va">MSEA_res</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">Multi_cond_res</span> <span class="op">=</span> <span class="va">Multi_cond_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>For detailed information about the output, check out documentation
<code><a href="../reference/Run_bootstrap_MSEA.html">?Run_bootstrap_MSEA</a></code>.</p>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<div class="section level3">
<h3 id="multi-condition-heatmap">Multi-condition Heatmap<a class="anchor" aria-label="anchor" href="#multi-condition-heatmap"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_MSEA_Multi_cond.html">plot_MSEA_Multi_cond</a></span><span class="op">(</span>combined_MSEA_res <span class="op">=</span> <span class="va">Multi_cond_res</span>,</span>
<span>                         alpha_cutoff <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/MSEA_multicond.jpg" style="width:100.0%"></p>
</div>
<div class="section level3">
<h3 id="barplot">Barplot<a class="anchor" aria-label="anchor" href="#barplot"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/barplot_MSEA_boot.html">barplot_MSEA_boot</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">MSEA_results</span><span class="op">[[</span><span class="st">"F"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                  q.value.cutoff <span class="op">=</span> <span class="fl">0.2</span>, by.statistic <span class="op">=</span> <span class="st">"ES"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/MSEA_bar.jpg" style="width:100.0%"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Bishoy Wadie, Martijn Molenaar, Theodore Alexandrov.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
